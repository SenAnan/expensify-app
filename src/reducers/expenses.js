const expensesDefaultState = [];

const expensesReducer = (state = expensesDefaultState, action) => {
	switch (action.type) {
		case 'ADD_EXPENSE':
			return [...state, action.expense];
		case 'REMOVE_EXPENSE':
			return state.filter((el) => el.id != action.id);
		case 'EDIT_EXPENSE':
			return state.map((el) => {
				if (el.id == action.id) {
					return {
						...el,
						...action.updates,
					};
				} else {
					return el;
				}
			});
		case 'SET_EXPENSES':
			return action.expenses;
		default:
			return state;
	}
};

export default expensesReducer;
