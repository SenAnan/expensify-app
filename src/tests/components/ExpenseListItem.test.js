import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

const expense = expenses[0];

test('should render ExpenseListItem', () => {
	const wrapper = shallow(<ExpenseListItem {...expense} />);
	expect(wrapper).toMatchSnapshot();
});
