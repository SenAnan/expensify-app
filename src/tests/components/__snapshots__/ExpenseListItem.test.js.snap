// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <[object Object]
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </[object Object]>
  <p>
    Amount: 
    $1.95
  </p>
  <p>
    Created at: 
    January 1st, 1970
  </p>
</div>
`;
